### ISSUE TITLE
Nullable properties in ViewModel are not fit for `equals`

### ISSUE THREAD

**Issue by kika** [marcglasberg/async_redux#103]
This doesn't work:
```dart
class _ViewM extends Vm {
  final String? error;
  final String name;
  _ViewM({required this.name, this.error}): super(equals: [name, error]);
}
```

Looking at `_listEquals` in `view_model.dart` I don't see anything preventing it from being `(List<T?>? list1, List<T?>? list2)`

Will PR be accepted or I'm breaking some architectural principles here?
(I believe `_listEquals` just requires the declaration change, but `_propsHashCode` will need to be guarded against null values

**Comment by dluksza** [comment]
Have you tried `equals: [name, error ?? Null]`? This should at least compile, don't know if it will work correctly since I'm still migrating my app to null safety. Hope this helps.

**Comment by marcglasberg** [comment]
@kika Could you please check if this is fixed for version [9.0.7]?

**Comment by marcglasberg** [comment]
@kika Closing this, please let me know if the problem is not solved for some reason. Thanks.
